js/data/line_bucket.js

```diff
@@ -8,22 +9,93 @@ module.exports = LineBucket;
  * @class LineBucket
  * @private
  */
-function LineBucket(buffers, layoutProperties) {
+function LineBucket(buffers, layoutDeclarations,  _, zoom) {
+
     this.buffers = buffers;
-    this.elementGroups = new ElementGroups(buffers.lineVertex, buffers.lineElement);
-    this.layoutProperties = layoutProperties;
+
+    this.layoutDeclarations = layoutDeclarations;
+    this.zoom = zoom;
 }
 
+LineBucket.prototype.calculatePaintAttributeOffsets = function() {
+    this.partiallyEvaluated = {};
+    var itemSize = 8;
+
+    var layerOffsets = {};
+
+    for (var l = 0; l < this.layers.length; l++) {
+        var layerID = this.layers[l];
+        var offsets = layerOffsets[layerID] = {};
+        var partiallyEvaluated = this.partiallyEvaluated[layerID] = {};
+        var paintDeclarations = this.layerPaintDeclarations[layerID];
+
+        var lineColor = paintDeclarations['line-color'];
+        if (lineColor && !lineColor.calculate.isFeatureConstant) {
+            offsets.color = itemSize;
+            itemSize += 4;
```

Comment:
I'm using 4 bytes for each attribute so that they are all aligned to 4 byte boundaries. Misaligned data can have a negative performance impact. Having gaps probably also has a performance impact. The only we can know is profiling. In the past this has been more noticeable on mobile devices I think.